src/editor/components/iomd-editor.jsx

```diff
@@ -3,227 +3,219 @@ import { connect } from "react-redux";
-    while (tprop.type === "property") {
-      tprop = getToken(cm, Pos(cur.line, tprop.start));
-      if (tprop.string !== ".") return; // don't do anything.
-      tprop = getToken(cm, Pos(cur.line, tprop.start));
-      if (!context) context = [];
-      context.push(tprop);
+    if (delimLines.join(",") !== prevProps.delimLines.join(",")) {
+      const newDecorations = delimLines.map(line => ({
+        range: new monaco.Range(line, 1, line, 1),
+        options: {
+          isWholeLine: true,
+          className: ".iomd-delim-line",
+          stickiness: decorationsDontGrow
+        }
+      }));
+      this.editor.deltaDecorations(iomdDelimLineDecorationIds, newDecorations);
     }
 
-    // remove all functions to allow message passing to eval-frame.
-    token.state.tokenize = undefined;
-    delete token.state.cc;
-    context.forEach(c => {
-      const cc = c;
-      cc.state.tokenize = undefined;
-      delete cc.state.cc;
-    });
+    this.editor.updateOptions({ wordWrap });
+  }
 
-    const from = Pos(cur.line, token.start);
-    const to = Pos(cur.line, token.end);
-
-    // the eval-frame, after receiving this message,
-    // generates the completion suggestions and sends back
-    // to be rendered by the CodeMirror instsance saved in
-    // window.ACTIVE_EDITOR_REF.
-    // After the .showHint function is called, we set window.ACTIVE_EDITOR_REF = undefined.
-    postMessageToEvalFrame("REQUEST_AUTOCOMPLETE_SUGGESTIONS", {
-      token,
-      context,
-      from,
-      to
-    });
-  };
+  handleEditorUpdate(content) {
+    this.props.updateIomdContent(content);
+    this.props.updateAutosave();
+  }
 
   render() {
-    // FIXME: restore autocomplete to working order
-    // const editorOptions = Object.assign(
-    //   {},
-    //   this.props.editorOptions,
-    //   {
-    //     extraKeys: {
-    //       'Ctrl-Space': this.props.codeMirrorMode === 'javascript' ?
-    // this.autoComplete : undefined,
-    //     },
-    //   },
-    // )
-
-    const { editorCursorLine, editorCursorCol } = this.props;
     return (
-      <ReactCodeMirror
-        editorDidMount={this.storeEditorInstance}
-        cursor={{ line: editorCursorLine, ch: editorCursorCol }}
-        value={this.props.content}
-        options={this.props.editorOptions}
-        onBeforeChange={this.updateIomdContent}
-        onCursorActivity={this.updateCursor}
-        style={{ height: "100%" }}
+      <div
+        ref={this.containerDivRef}
+        style={{ width: "100%", height: "100%" }}
       />
     );
   }
 }
 
 function mapStateToProps(state) {
-  const codeMirrorMode = "iomd";
-  const editorOptions = {
-    mode: codeMirrorMode,
-    lineWrapping: false,
-    matchBrackets: true,
-    autoCloseBrackets: true,
-    theme: "eclipse",
-    autoRefresh: true,
-    lineNumbers: true,
-    keyMap: "sublime",
-    comment: true
-  };
+  const wordWrap = state.wrapEditors ? "on" : "off";
```

Comment:
i think mapStateToProps is for translating data in the redux store to the data needed for a component so that you can have less logic in the component

Reply:
In general I agree, though the fact that monaco expects a parameter with a string argument "on"/"off" feels like an implementation detail that is best dealt with inside the thing that's talking to monaco.